.github/workflows/linux-build.yaml

```diff
@@ -0,0 +1,43 @@
+name: Linux Build
+on: [workflow_dispatch, pull_request]
+
+jobs:
+  build:
+    name: Build
+    strategy:
+      matrix:
+        os: [large-ubuntu-22.04-xxl]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Setup cache
+        uses: actions/cache@v3
+        with:
+          path: cache
+          key: compile-${{ github.ref }}-${{ github.sha }}
```

Comment:
Looks like this is based on the workflow from everest-core like two weeks ago.  The caching is not really working well here in cases where one has `-` in the branch name, which is not uncommon.  See the current workflow file in everest core on how to fix that